server/methods/core/orders.js

```diff
@@ -139,6 +139,18 @@ Meteor.methods({
       throw new Meteor.Error(403, "Access Denied");
     }
 
+    // Server-side check to make sure discount is not greater than orderTotal.
+    let orderTotal =
+      order.billing[0].invoice.subtotal
+      + order.billing[0].invoice.shipping
+      + order.billing[0].invoice.taxes;
+
+    if (discount > orderTotal) {
+      const err = "Discount is greater than the cart total";
+      Logger.error(err);
```

Comment:
I'm not sure we need to log this error here at least not at this level since it's just a user error that we handle. There is nothing an admin can/needs-to do